### PR TITLE
Enable DefaultAzureCredential authentication for Azure filesystem block

### PR DESCRIPTION
**Pull request by samdyzon**
# Background

Recent updates to the Azure filesystem block introduced the ability to authenticate with Azure via a service principal (#6838). This short PR extends the parameters of the Azure filesystem block by exposing the `anon` parameter of the ALDFS filesystem class which is the main parameter required to tell ALDFS to use the `DefaultAzureCredential` authentication mechanism.

[DefaultAzureCredential](https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) attempts to automatically determine what type of credentials to use based on the execution environment, and enables authentication via the following:

- A service principal configured by environment variables.
- An Azure managed identity. 
- The user currently signed in to Visual Studio Code.
- The identity currently logged in to the Azure CLI.
- The identity currently logged in to Azure PowerShell.

## Problem Statement

ALDFS exposes a single parameter, `anon`, with a default value of `True`. If a user provides no `credential`, `sas_token`, and `account_key`, and `anon` is `False` then ADLFS will use `DefaultAzureCredential`. See [the ADLFS spec.py](https://github.com/fsspec/adlfs/blob/main/adlfs/spec.py#L482) file for details.

## Solution

By exposing the `anon` parameter and leaving it default to True, this change is a backwards-compatible method for enabling additional authentication mechanisms that are currently impossible with the Azure filesystem block, including:

- Deploying flows to Azure Blob storage with an appropriate Azure AD user logged in to the CLI
- Running flows in Kubernetes using the [Azure AD Workload Identity operator](https://azure.github.io/azure-workload-identity/docs/)
- Deploying/running flows on an Azure VM backed by a Managed Identity

## Notes

I've looked at PR #6838 as a reference for the changes in this PR, the only thing I don't know how to do is to update the block test hash. I was also unable to test this change locally because attempting to run `prefect block register -m prefect.filesystems` throws an error about modifying fields. 

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Review by desertaxle — APPROVED** [review]

File: docs/concepts/filesystems.md [hunk 1]
```diff
@@ -104,6 +104,7 @@ The `Azure` file system block enables interaction with Azure Datalake and Azure
 | azure_storage_tenant_id | Azure storage tenant ID. |
 | azure_storage_client_id | Azure storage client ID. |
 | azure_storage_client_secret | Azure storage client secret. |
+| azure_storage_anon | Anonymous authentication, disable to use `DefaultAzureCredential` |
```
Comment (tpdorsey) [inline docs/concepts/filesystems.md:107]:
```suggestion
| azure_storage_anon | Anonymous authentication, disable to use `DefaultAzureCredential`. |
```

**Review by tpdorsey — APPROVED** [review]

**Comment by samdyzon** [comment]
Hi folks, is there anything you need me to do to have this merged?

**Comment by OliverKleinBST** [comment]
Hi @madkinsz, @samdyzon 

we are also very much interested to using service principal, so would be very much interested in getting this ticket integrated.
Is there anything we could help with to make it happen?

**Comment by madkinsz** [comment]
Hi @samdyzon! Looks like this got a little lost.

Is it feasible to add a test covering this change? Can you resolve the merge conflicts with `main` or do you need help?

**Review by madkinsz — APPROVED** [review]

**Comment by samdyzon** [comment]
Hi @madkinsz, apologies - all of these notifications came through in the middle of the night for me, but I'm glad this has been merged. Thanks!

**Pull request merged**